packages/firestore/src/api/database.ts

```diff
@@ -420,7 +420,9 @@ export class Firestore implements firestore.FirebaseFirestore, FirebaseService {
     const persistenceKey = IndexedDbPersistence.buildStoragePrefix(
       this.makeDatabaseInfo()
     );
-    return IndexedDbPersistence.clearPersistence(persistenceKey);
+    return this._queue.enqueue(() => {
```

Comment:
If clearPersistence() fails this will cause the async queue to fail also, and leave the system in an unusable state.

You need to use a similar technique to what we did on Android to prevent that here.

`Deferred` in util/promise is the best way to do that.